lib/matplotlib/cbook.py

```diff
@@ -1505,6 +1505,14 @@ def issubclass_safe(x, klass):
 
 def safe_masked_invalid(x, copy=False):
     x = np.array(x, subok=True, copy=copy)
+    if not x.dtype.isnative:
+        if copy:
+            # if a copy, swap in place
+            x.byteswap(True)
+            x = x.newbyteorder()
+        else:
+            # otherwise make a swapped copy
+            x = x.byteswap(False).newbyteorder()
```

Comment:
Why is this if/else needed?  I think that `x = x.byteswap(copy).newbyteorder()` will suffice.  The `byteswap` method returns a new array if copy is False and a view into the (modified) original if copy (inplace) is True.
```
In [31]: xm = np.ma.ones((2,))

In [32]: xm1 = xm.byteswap(False).newbyteorder()

In [33]: xm2 = xm.byteswap(True).newbyteorder()

In [34]: (xm1 == xm2).all()
True

In [35]: xm1.dtype
dtype('>f8')

In [36]: xm2.dtype
dtype('>f8')
```